src/main/java/io/kestra/plugin/langchain4j/embeddings/MongoDBAtlas.java

```diff
@@ -0,0 +1,151 @@
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+import org.awaitility.Awaitility;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Getter
+@SuperBuilder
+@NoArgsConstructor
+@Plugin(beta = true)
+@JsonDeserialize
+@Schema(
+    title = "MongoDB Atlas Embedding Store"
+)
+public class MongoDBAtlas extends EmbeddingStoreProvider {
+
+    @Schema(title = "The username")
+    private Property<String> username;
+
+    @Schema(title = "The password")
+    private Property<String> password;
+
+    @NotNull
+    @Schema(title = "The scheme (e.g. mongodb+srv)")
+    private Property<String> scheme;
+
+    @NotNull
+    @Schema(title = "The host")
+    private Property<String> host;
+
+    @Schema(title = "The database")
+    private Property<String> database;
+
+    @Schema(title = "The connection string options")
+    private Property<Map<String, Object>> options;
+
+    @NotNull
+    @Schema(title = "The collection name")
+    private Property<String> collectionName;
+
+    @NotNull
+    @Schema(title = "The index name")
+    private Property<String> indexName;
+
+    @Schema(title = "The metadata field names")
+    private Property<List<String>> metadataFieldNames;
+
+    @Schema(title = "Create the index")
+    private Property<Boolean> createIndex;
+
+    @Override
+    public EmbeddingStore<TextSegment> embeddingStore(RunContext runContext, int dimension, boolean drop) throws IOException, IllegalVariableEvaluationException {
+
+        var mongoClient = MongoClients.create(buildUri(runContext));
+
+        var renderedCreateIndex = runContext.render(createIndex).as(Boolean.class).orElse(false);
+        var store = MongoDbEmbeddingStore.builder()
+            .fromClient(mongoClient)
+            .databaseName(runContext.render(database).as(String.class).orElseThrow())
+            .collectionName(runContext.render(collectionName).as(String.class).orElseThrow())
+            .indexName(runContext.render(indexName).as(String.class).orElseThrow())
+            .createIndex(renderedCreateIndex)
+            .indexMapping(
+                metadataFieldNames != null ?
+                    IndexMapping.builder()
+                        .dimension(dimension)
+                        .metadataFieldNames(new HashSet<>(runContext.render(metadataFieldNames).asList(String.class)))
+                        .build() : null
+            )
+            .build();
+
+        if (renderedCreateIndex) {
+            // Creating a vector search index can take up to a minute, so this delay allows the index to become queryable
+            Awaitility.await()
+                .atMost(Duration.ofMinutes(1))
+                .pollInterval(Duration.ofSeconds(3))
```

Comment:
I think by default it will do an exponential retry which would be better that using a fixed interval.
Can you remove pollInterval? Or set it to a lower value as 1s.